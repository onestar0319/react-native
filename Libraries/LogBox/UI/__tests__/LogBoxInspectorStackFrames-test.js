/**
 * Copyright (c) Facebook, Inc. and its affiliates.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 *
 * @format
 * @emails oncall+react_native
 * @flow
 */

'use strict';

const React = require('react');
const LogBoxInspectorStackFrames = require('../LogBoxInspectorStackFrames')
  .default;
const LogBoxLog = require('../../Data/LogBoxLog').default;
const render = require('../../../../jest/renderer');

const log = new LogBoxLog(
  {
    content: 'Some kind of message (latest)',
    substitutions: [],
  },
  [
    {
      column: 1,
      file: 'dependency.js',
      lineNumber: 1,
      methodName: 'dep',
      collapse: true,
    },
    {
      column: 1,
      file: 'app.js',
      lineNumber: 1,
      methodName: 'foo',
      collapse: false,
    },
  ],
  'Some kind of message (latest)',
  [],
  false,
);

describe('LogBoxInspectorStackFrame', () => {
  it('should render stack frames with 1 frame collapsed', () => {
    const output = render.shallowRender(
      <LogBoxInspectorStackFrames onRetry={() => {}} log={log} />,
    );

    expect(output).toMatchSnapshot();
  });
});
