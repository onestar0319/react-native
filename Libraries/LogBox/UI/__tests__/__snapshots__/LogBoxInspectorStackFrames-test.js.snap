// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LogBoxInspectorStackFrame should render stack frames with 1 frame collapsed 1`] = `
<View
  style={
    Object {
      "marginTop": 15,
    }
  }
>
  <StackFrameHeader
    onRetry={[Function]}
    status="NONE"
  />
  <View
    style={
      Object {
        "paddingBottom": 10,
      }
    }
  >
    <StackFrameList
      list={
        Array [
          Object {
            "collapse": false,
            "column": 1,
            "file": "app.js",
            "lineNumber": 1,
            "methodName": "foo",
          },
        ]
      }
      status="NONE"
    />
    <StackFrameFooter
      message="See 1 more frames"
      onPress={[Function]}
    />
  </View>
</View>
`;
